### PR TITLE
feat(types): switch from custom types to api-snippets defined types

### PR DESCRIPTION
**Pull request by lewisdaly**
* switches from types defined in sdk to api-snippets
* updates the thirdparty api to use api-snippets
* replaces old API calls with updated calls (e.g. `PATCH /consentRequests/{id}`)

part of mojaloop/project#2263

### CODE REVIEW

File: src/interface/api_template/components/schemas/Participant.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+title: Participant
```
Comment (kleyow) [inline src/interface/api_template/components/schemas/Participant.yaml:1]: hmm. you shouldn't need to rewrite these schemas if they are apart of the fspiop/thirdparty and in snippets. is there some schema mismatch that I'm not seeing?

File: src/interface/api_interfaces/openapi.d.ts [hunk 2]
```diff
@@ -0,0 +1,1698 @@
+/**
```
Comment (kleyow) [inline src/interface/api_interfaces/openapi.d.ts:1]: you shouldn't need to regenerate data types if they are apart of fspiop/thirdparty.

Comment (lewisdaly) [reply src/interface/api_interfaces/openapi.d.ts:1]: Ah ok, yeah I don't think we are referring to this file, so can probably just delete it, and remove it from the build task.

File: src/interface/api_template/components/schemas/Participant.yaml (hunk 1 above)
Comment (lewisdaly) [reply src/interface/api_template/components/schemas/Participant.yaml:1]: Yeah this was leftover from Sridhar's original work updating the types and moving from what our interns had already written. I suspect I should be able to delete a lot of these components and update them to references to api-snippets.

I'll take care of that in a followup PR (see the todos in `mojaloop-template.yaml`)

File: src/server/handlers/firestore/transactions.ts [hunk 3]
```diff
@@ -157,7 +158,7 @@ async function handleAuthorization(server: StateServer, transaction: Transaction
 
   // TD - LD eww so much messy casting going on
   // @ts-ignore - todo fix me!
-  const requestBody: PutThirdpartyRequestsTransactionsAuthorizationsRequest = {
+  const requestBody: tpAPI.Schemas.ThirdpartyRequestsTransactionsIDAuthorizationsPutResponsequest = {
```
Comment (kleyow) [inline src/server/handlers/firestore/transactions.ts:161]: typo?

Comment (lewisdaly) [reply src/server/handlers/firestore/transactions.ts:161]: Haha yeah I guess the `ts-ignore` should be removed and we would see a built time error.

**Review by kleyow — APPROVED** [review]

**Pull request merged**